AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Operations/Helpers/FetchAuthSessionOperationHelper.swift

```diff
@@ -39,6 +41,7 @@ struct FetchAuthSessionOperationHelper {
                     if tokens.doesExpire(in: Self.expiryBufferInSeconds) {
                         let event = AuthorizationEvent(eventType: .refreshSession)
                         authStateMachine.send(event)
+                        self.listenForSession(authStateMachine: authStateMachine, completion: completion)
```

Comment:
Added the call to listen in scenarios where refreshing has been started, and the fetch auth session completion needs to be determined.

Reply:
Thanks for adding this